server/tinylicious/src/routes/ordering/documents.ts

```diff
@@ -72,5 +81,33 @@ export function create(storage: IDocumentStorage): Router {
 		);
 	});
 
+	/**
+	 * Passes on content to all clients in a collaboration session happening on the document via means of signal.
+	 */
+	router.post("/:tenantId/:id/broadcast-signal", (request, response) => {
+		const tenantId = getParam(request.params, "tenantId");
+		const documentId = getParam(request.params, "id");
+		const signalContent = getParam(request.body, "signalContent");
+		const documentP = storage.getDocument(
+			getParam(request.params, "tenantId"),
+			getParam(request.params, "id"),
+		);
+		documentP.then(
+			(_document: any) => {
+				try {
+					const signalRoom: IRoom = { tenantId, documentId };
+					const payload: IBroadcastSignalEventPayload = { signalRoom, signalContent };
+					collaborationSessionEventEmitter.emit("broadcast-signal", payload);
+					response.status(200).send("OK");
+				} catch (error) {
+					response.status(500).send(error);
+				}
+			},
+			(error) => {
+				response.status(400).json(error);
+			},
+		);
```

Comment:
Did we try these already? I'm not sure that doing anything with `response` inside a Promise that is not awaited nor returned will actually matter. With the current code I kind of expect this route to always return 200 immediately, regardless of what happens with the signal (but it makes sense that it'd at least return 500 if something went wrong).